packages/openapi3-parser/test/unit/parser/parseObject-test.js

```diff
@@ -141,6 +144,13 @@ describe('#parseObject', () => {
         "'Example Object' is missing required property 'required1'",
         "'Example Object' is missing required property 'required2'",
       ]);
+
+      // assert errors can be frozen (we're testing that the source map
+      // for object was cloned instead of referenced)
```

Comment:
Out of curiosity how would it look like without the clone? It is not completely obvious for me how this is testing the not-a-reference.

Reply:
`freeze()` would throw the following error:

> Cannot assign to read only property 'parent' of object '#<ArrayElement>'

Reply:
That’s because effectively it’s doing this:

```
freeze() {
  freeze “this”
  
  for element in (all meta, attributes, content):
    element.parent = this
    element.freeze()
}
```